features/dump_offline_inventories.lua

```diff
@@ -22,59 +28,54 @@ local function spawn_player_corpse(player, banned, timeout_minutes)
     local player_index = player.index
     offline_player_queue[player_index] = nil
 
-    if not banned and player.connected then
-        return
-    end
-
-    local inv_main = player.get_inventory(defines.inventory.character_main)
-    local inv_trash = player.get_inventory(defines.inventory.character_trash)
-
-    local inv_main_contents
-    if inv_main and inv_main.valid then
-        inv_main_contents = inv_main.get_contents()
-    end
-
-    local inv_trash_contents
-    if inv_trash and inv_trash.valid then
-        inv_trash_contents = inv_trash.get_contents()
-    end
+    local inventory_types = {
+        defines.inventory.character_main,
+        defines.inventory.character_guns,
+        defines.inventory.character_ammo,
+        defines.inventory.character_vehicle,
+        defines.inventory.character_trash,
+    }
 
-    local inv_corpse_size = 0
-    if inv_main_contents then
-        inv_corpse_size = inv_corpse_size + (#inv_main - inv_main.count_empty_stacks())
+    if banned or game.tick < config.startup_gear_drop_hours * HOUR_TO_TICKS or Rank.less_than(player.name, Ranks.regular) then
```

Comment:
Just because I don't like having to think about order of operations when reading code.
```suggestion
    if banned or game.tick < (config.startup_gear_drop_hours * HOUR_TO_TICKS) or Rank.less_than(player.name, Ranks.regular) then
```

Reply:
Np